lucene/core/src/java/org/apache/lucene/geo/XYEncodingUtils.java

```diff
@@ -33,11 +33,12 @@
   private XYEncodingUtils() {
   }
 
-  /** validates value is within +/-{@link Float#MAX_VALUE} coordinate bounds */
-  public static void checkVal(double x) {
-    if (Double.isNaN(x) || x < MIN_VAL_INCL || x > MAX_VAL_INCL) {
+  /** validates value is a number and finite */
+  static float checkVal(float x) {
+    if (Float.isNaN(x) || Float.isInfinite(x)) {
```

Comment:
This could be a single test.
```suggestion
    if (Float.isFinite(x) == false) {
```